### ISSUE TITLE
does "egress" configuration for pipeline means that putting files into /pfs/out will not consume disk space?

### ISSUE THREAD

**Issue by aka-toxa** [pachyderm/pachyderm#4247]
Hi 

I have this question because it's not super clear from the documentation.

The documentation said that 

```
egress enables you to push the results of a pipeline to an external datastore such as Amazon S3, Google Cloud Storage, or Azure Blob Storage
```

Does that mean that all files from output directory will live only on specified store and not on the disk attached to the pod?

I wonder because I have a case where I built a data processing pipeline that lives forever and waits until new files arrived in the input repo, I have hundreds of thousands of files that should be processed by this pipeline one by one and I wonder how can I delete processed files to not consume node's disk with old, already processed files. I've already managed to configure lazy input to not consume disk space by input files and now I faced with the issue where pachyderm folder grows because of the output folder

**Comment by aka-toxa** [comment]
it's on azure

hmm maybe I deployed things wrong because it eats disk attached to the pod that apparently is a node disk

**Comment by JoeyZwicker** [comment]
Did you do `pachctl deploy azure` or `deploy local` to an azure instance?

**Comment by aka-toxa** [comment]
I did 

```
helm upgrade pachyderm stable/pachyderm --tls --version 0.1.7 --install --wait --recreate-pods --reset-values --namespace pachyderm -f <(envsubst < $(dirname $BASH_SOURCE)/pachydermConfig.yaml)
```

where pachydermConfig.yaml is

```
rbac:
  create: true
credentials: "microsoft"
microsoft:
  container: "" # Blob Container name
  id: "" # Storage account name
  secret: "" # Storage account access key
pachd:
  replicaCount: 1
  pfsCache: 4G
  resources:
    requests:
      cpu: 1000m
      memory: 2G
etcd:
  persistence:
    storageClass: managed-premium
    enabled: true
    size: 10G
    accessMode: ReadWriteOnce
  resources:
    requests:
      cpu: 1000m
      memory: 2G
```

did you mean that `pfs/out` folder is mounted from the azure blob, not from disk? it's strange, because when I give 100TB of data to pachyderm it shutdown the cluster because it eats whole disk.

**Comment by jdoliner** [comment]
It's possible that you're overwelming disk space by having several datums queued up for processing on the same node. How many datums is this 100TB broken up into and how many nodes is the processing of those datums split between? As @JoeyZwicker said data never gets stored permanently on disk if you're using blob storage, but it's possible for there to be enough data in flight that it puts disk pressure on the nodes.

**Comment by aka-toxa** [comment]
just did the test in the different isolated environments with a clean install and it looks like with lazy input pachyderm indeed doesn't use the disk at all. I've done something horribly wrong with my production setup. @JoeyZwicker Thanks for the clarification about output folder 

@jdoliner it looks like I did fill up the disk with input files when I didn't turn lazy option in my pipeline spec. When I did a small test in a separate environment I didn't seen any data overwhelming on the disk

I guess that we can close the ticket now. Thanks.
